### ISSUE TITLE
Connect with cyrillic characters in database path | windows

### ISSUE THREAD

**Issue by VladDB** [hipercompl/fbdb#3]
Hi! Thank you for package, it's very good!
But when I try to connect to the database using attach and path with Cyrillic characters in windows , it freezes waiting for first from resultPort in _askWorker. If I try the same thing in linux, everything works correct.

**Comment by ttyrakow** [comment]
Thanks for reporting this. I'll try to replicate this scenario and pinpoint the cause. I've tested it with Polish diacritics in the database path, but not cyrillic. Just for clarification: when you try to connect from Windows to the same database, but using other tools (like the standard Firebird isql console or Flamerobin), does it work?

**Comment by ttyrakow** [comment]
Done some testing with the native `isql` tool from Firebird. It looks like it's a problem with the actual Firebird client library to accept database paths with cyrillic characters in them:
```
isql -user sysdba -password masterkey C:\db\четверг\employee.fdb
Statement failed, SQLSTATE = 08001
I/O error during "CreateFile (open)" operation for file "C:\DB\???????\EMPLOYEE.FDB"
-Error while trying to open file
```
I'm afraid I can't do much about it. Nevertheless, fbdb should not freeze if such thing happens and that's something I can definitely improve (an exception should be thrown in that case).

**Comment by ttyrakow** [comment]
It looks like the Linux version works because the connection character set (it's UTF-8 in fbdb) matches the system-wide character set of the system where the server is running. On Windows, I assume you have WIN-1251, and the UTF-8 encoded path is not correctly decoded by the server. See also here: [https://stackoverflow.com/questions/41567466/firebird-connection-using-non-ascii-characters-in-path](https://stackoverflow.com/questions/41567466/firebird-connection-using-non-ascii-characters-in-path). It's probably safest to have just ASCII characters in DB paths.
However, my remark about correcting fbdb behavior in this scenario is still valid and I intend to take care of it.

**Comment by VladDB** [comment]
I thought about this and guess too that problem the encoding. I used DBeaver to connect to the database and it does not throw any exceptions. But isql showed me the same problem. I also tested connecting using the SOCI library for C++, and in that case I had the same problem as with isql, before I didn't convert the path to WIN-1251 from UTF-8. 

Thanks so much for your feedback!

**Comment by ttyrakow** [comment]
The issue was addressed in version 1.0.2 (published at pub.dev on 2024-09-16). The errors containing malformed UTF-8 strings are properly handled now. The actual problem was that Firebird's IUtil interface was not able to properly format an error message, because it contained invalid UTF-8 sequences (the actual cyrillic characters from the database path). So, the fixed problem is actually only related to the **handling** of the error. If the Firebird client is not able to properly pass the DB path to the server, there's nothing I can do about it on the Dart level (the client code doesn't know the system character encoding of the server host, so there's no way to transcode the DB path properly).
The general solution to the reported issue is not to use non-ASCII characters in DB paths or to configure DB aliases on the server (and use the alias name instead of a hardcoded path to a FDB file, which is probably a good idea anyway).
